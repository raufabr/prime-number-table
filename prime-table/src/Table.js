import "./Table.css";
import ReactHtmlParser from "react-html-parser";
import { findNPrimes } from "./primes";
export const Table = ({ input }) => {
	const generateTable = (primeNums) => {
		let output = "";
		for (let i = 0; i <= primeNums.length - 1; i++) {
			output += "<tr>" + "<th>" + primeNums[i] + "</th>";
			for (let j = 0; j <= primeNums.length - 1; j++) {
				output += "<td>" + primeNums[i] * primeNums[j] + "</td>";
			}
			output += "</tr>";
		}

		return ReactHtmlParser(output);
	};

	return (
		<table>
			<thead>
				<tr>
					<th />
					{findNPrimes(input).map((num) => (
						<th>{num}</th>
					))}
				</tr>
			</thead>
			<tbody>{generateTable(findNPrimes(input))}</tbody>
		</table>
	);
};
