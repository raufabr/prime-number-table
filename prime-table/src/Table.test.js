import { Table } from "./Table";
import { render } from "@testing-library/react";

describe("test Table", () => {
	const inputRows = 3;
	const { queryAllByRole } = render(<Table />);

	test("It renders a table", () => {
		expect(queryAllByRole("table")).toHaveLength(1);
	});
	// test("It has the right number of rows", () => {
	// 	expect(queryAllByRole("row")).toHaveLength(inputRows + 1);
	// });
});
