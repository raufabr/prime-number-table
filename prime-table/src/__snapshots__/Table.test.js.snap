// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that findNPrimes returns the right elements in array matches snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <th />
        <th>
          2
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th>
          2
        </th>
        <td>
          4
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <th />
        <th>
          2
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th>
          2
        </th>
        <td>
          4
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
